### PR TITLE
Refactor omero-prometheus-tools

### PR DESCRIPTION
**Pull request by manics**
Replaces https://github.com/IDR/omero-prometheus-tools/pull/2

- Update the omero session metrics with user and group counts
- Pip installable
- Login once on startup instead of for each request
- Exit if an error occurs so it's clear to the user (or systemd) that there is a problem
- Add travis (needs to be enabled on this repo)

Testing: `omero-prometheus-sessions.py -s omero.example.org -u public -w public`. See the updated top-level README.md

Tag: `0.1.0`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    rm -rf /var/cache/yum && \
+    pip install https://github.com/openmicroscopy/zeroc-ice-py-centos7/releases/download/0.0.3/zeroc_ice-3.6.4-cp27-cp27mu-linux_x86_64.whl
+
+RUN pip install omego && \
+    mkdir -p /opt/omero && \
+    cd /opt/omero && \
+    omego download python --sym OMERO.py && \
+    echo /opt/omero/OMERO.py/lib/python > /usr/lib/python2.7/site-packages/omero.pth
+
+RUN pip install https://github.com/manics/omero-prometheus-tools/archive/pip-package.zip
```
Comment (joshmoore) [inline Dockerfile]: This seems odd. Shouldn't this be building the source in the current tree?

**Review by joshmoore — COMMENTED** [review]
When this is pip-installed how is it used? The service files seem to assume it's under /opt/o-p-t

File: Dockerfile (hunk 1 above)
Comment (manics) [reply Dockerfile]: Could do

**Comment by manics** [comment]
You can just pip install it and run it directly (see description). The service file is for convenience if you want to run it with systemd. Alternative is to get rid of it and create it in a playbook?

**Comment by kennethgillen** [comment]
Where does the monitoring user's password sit on-disk, and with what filesystem-level permissions?

**Comment by manics** [comment]
user/pass can be stored in a separate config file  or startup script managed elsewhere

**Comment by manics** [comment]
Something to consider: worth merging these two separate `bin/` scripts into a single script?

**Comment by manics** [comment]
I've combined them, set a default port of 9449, added to https://github.com/prometheus/prometheus/wiki/Default-port-allocations

**Review by joshmoore — COMMENTED** [review]
Generally looks good for a 0.1.0. A few questions:

 * I'm not sure which 2 scripts you're referring to.
 * Objections to migrating to github.com/ome?
 * Objections to holding off on the pypi package?

**Comment by manics** [comment]
> I'm not sure which 2 scripts you're referring to.

Previously there were separate scripts for the two groups of metrics (sessions and object counts). I've combined them.

> Objections to migrating to github.com/ome?

:+1:

> Objections to holding off on the pypi package?

:+1: I'm happy if we never put this on pypi since it'll be deployed with a separate ansible role and it avoids another manual step in the release process

If you're happy can you tag and transfer?

**Comment by joshmoore** [comment]
Done: https://github.com/ome/omero-prometheus-tools/releases/tag/0.1.0

**Pull request merged**